flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/schema/TableSourceTable.scala

```diff
@@ -61,11 +64,33 @@ class TableSourceTable[T](
       tableIdentifier.getObjectName),
     statistic) {
 
+  def this(relOptSchema: RelOptSchema,
+    tableIdentifier: ObjectIdentifier,
+    rowType: RelDataType,
+    statistic: FlinkStatistic,
+    tableSource: TableSource[T],
+    isStreamingMode: Boolean,
+    catalogTable: CatalogTable) = this(relOptSchema, tableIdentifier, rowType, statistic,
+    tableSource, isStreamingMode, catalogTable, Collections.emptyMap())
+
   Preconditions.checkNotNull(tableSource)
   Preconditions.checkNotNull(statistic)
   Preconditions.checkNotNull(catalogTable)
 
-  override def getQualifiedName: JList[String] = explainSourceAsString(tableSource)
+  override def getQualifiedName: JList[String] = {
+    val sourceExplain = explainSourceAsString(tableSource)
+    if (dynamicOptions.size() == 0) {
+      sourceExplain
+    } else {
+      // Add the dynamic options as part of the table digest,
+      // this is a temporary solution, we expect to avoid this
+      // before Calcite 1.23.0.
```

Comment:
Create a issue to track this?

Reply:
Thanks, i have fired a issue https://issues.apache.org/jira/browse/FLINK-17147